src/Components/ServiceScene/ServiceScene.tsx

```diff
@@ -350,14 +353,70 @@ export class ServiceScene extends SceneObjectBase<ServiceSceneState> {
 
   private subscribeToFieldsVariable() {
     const fieldsVar = getFieldsVariable(this);
+
     return fieldsVar.subscribeToState((newState, prevState) => {
       if (!areArraysEqual(newState.filters, prevState.filters)) {
+        this.manageJsonParserProps(newState, prevState);
         this.state.$detectedFieldsData?.runQueries();
         this.state.$logsCount?.runQueries();
       }
     });
   }
 
+  /**
+   * If we removed a filter from the ad-hoc variables, we need to remove old json parser props
+   */
+  private manageJsonParserProps(newState: AdHocFiltersVariable['state'], prevState: AdHocFiltersVariable['state']) {
+    const lineFormatVariable = getLineFormatVariable(this);
+    if (!newState.filters.length && !lineFormatVariable.state.filters.length) {
+      clearJsonParserFields(this);
+
+      // A field was removed
+    } else if (newState.filters.length < prevState.filters.length) {
+      // If no other field filter has the same key, then we can remove the json parser
+      const filterDiff = prevState.filters.filter(
+        (prevFilter) => !newState.filters.find((newFilter) => newFilter.key === prevFilter.key)
+      );
+
+      if (filterDiff.length) {
+        filterUnusedJSONParserProps(this);
+      }
+
+      // A field was added
+    } else if (newState.filters.length > prevState.filters.length) {
+      const jsonVar = getJsonFieldsVariable(this);
+
+      // Gets any new filters that were added
+      const newFilters = newState.filters.filter(
+        (newFilter) => !prevState.filters.find((prevFilter) => newFilter.key === prevFilter.key)
+      );
+
+      const jsonParserPropsToAdd: AdHocFilterWithLabels[] = [];
```

Comment:
The "prop" keyword also tripped me out, even having to look for what are the "detected field props". Here we're syncing filters, right? Because later we add json parser props to the filter state of the jsonVar.

Reply:
Yeah I'm with you, it sucks because they are ad-hoc filters, but they are not logQL filters, Trevor and I settled on calling them JSON parser props, but ad-hoc filters makes everything messy. 
We can just refer to them as filters in the ad-hoc variable context and maybe it doesn't matter that they aren't filtering logs. 

Renamed: https://github.com/grafana/logs-drilldown/pull/1122/commits/0078fe712b4f1ddbc1f2e809470dbdd3e26f9502